--- meson-0.47.2.orig/mesonbuild/dependencies/dev.py
+++ meson-0.47.2/mesonbuild/dependencies/dev.py
@@ -219,7 +219,7 @@ class LLVMDependency(ConfigToolDependenc
             libdir = self.get_config_value(['--libdir'], 'link_args')[0]
 
             expected_name = 'libLLVM-{}'.format(self.version)
-            re_name = re.compile(r'{}.(so|dll|dylib)'.format(expected_name))
+            re_name = re.compile(r'{}.(so|dll|dylib)$'.format(expected_name))
 
             for file_ in os.listdir(libdir):
                 if re_name.match(file_):
